cmd/firmware-action/recipes/config.go

```diff
@@ -294,9 +299,21 @@ type FirmwareModule interface {
 	buildFirmware(ctx context.Context, client *dagger.Client) error
 }
 
-// ===========
-//  Functions
-// ===========
+// ===============================
+//  Functions for FirmwareModules
+// ===============================
+
+// FilenameForFirmwareModule is used to take a user-defined module name and make it into filename, removing
+// all problematic characters
+func FilenameForFirmwareModule(name string, extension string) string {
+	// For example:
+	//   "Coreboot Example" should return "Coreboot_Example.json"
+	return fmt.Sprintf("%s.%s", filesystem.Filenamify(name), extension)
```

Comment:
We already have the new `Filenamify` function, would be nice to sanity check the extension string as well then so that we don't accidentally input something like the whole bee movie script here.